music21/spanner.py

```diff
@@ -1275,10 +1317,12 @@ def getByClassIdLocalComplete(self, className, idLocal, completeStatus):
         return self.getByClass(className).getByIdLocal(
             idLocal).getByCompleteStatus(completeStatus)
 
-    def setPendingSpannedElementAssignment(
+    def setPendingFirstSpannedElementAssignment(
         self,
         sp: Spanner,
         className: str,
+        offsetInScore: OffsetQL,
+        staffKey: int
```

Comment:
these should be optional with sensible defaults to keep backwards compatible.  Trust me, there is at least one library that will break immediately if this is released without a backwards incompatible tag.

Reply:
The two new attributes are not discussed in the docs -- as a reader I would have no idea what setting them to anything but 0. and 0 are.  Give defaults of 0. and 0 and then add documentation of what it means to have another number besides them.  Also noticing that there are no tests on the new attributes.

What happens if I don't care about the next offset in score for the pending attribute?  Note that spanners exist in other formats than musicxml, so this change potentially breaks those parsers (most of which aren't tested nearly as well as musicxml).  So this is a big -1 for me unless it is optional.

Where is staffKey used?  It appears to be set and then never read.

Reply:
Does the MNX parser you're working on use these "pending" spanner APIs?  No other parser in music21 does.  

The non-use of staffKey is a bug (I just ran into it).  I _should_ be using staffKey (so that PartStaff splits put the spanner in the correct staff), and I am not.

I'm happy to default offsetInScore and staffKey to 0. and 0, respectively.

Reply:
Oh, actually, maybe they should both default to None, so the caller can say "I don't care", and freePending...() can ignore offsetInScore when looking for a match.

Reply:
Since I have now put the original setPendingSpannedElementAssignment/freePendingSpannedElementAssignment APIs back in place _without_ offsetInScore and staffKey, my new APIs (setPendingFirstSpannedElementAssignment and freePendingFirstSpannedElementAssignment) still require both offsetInScore and staffKey, and are always called with both from xmlToM21.py.